File: plate-be/src/main/java/tn/enicarthage/plate_be/services/SalleService.java
package tn.enicarthage.plate_be.services;

import tn.enicarthage.plate_be.dtos.SalleDTO;
import tn.enicarthage.plate_be.dtos.SalleRequestDTO;

import java.util.List;

public interface SalleService {

    List<SalleDTO> getAll();

    SalleDTO getById(String id);

    SalleDTO create(SalleRequestDTO dto);

    SalleDTO update(String id, SalleRequestDTO dto);

    void delete(String id);
}

File: plate-fe/src/app/core/services/salle.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Salle } from '../models/salle.model';

@Injectable({
  providedIn: 'root'
})
export class SalleService {
  private apiUrl = 'http://localhost:8080/api/salles';

  constructor(private http: HttpClient) {}

  getAll(): Observable<Salle[]> {
    return this.http.get<Salle[]>(this.apiUrl);
  }

  getById(id: string): Observable<Salle> {
    return this.http.get<Salle>(`${this.apiUrl}/${id}`);
  }

  create(salle: Salle): Observable<Salle> {
    return this.http.post<Salle>(this.apiUrl, salle);
  }

  update(id: string, salle: Salle): Observable<Salle> {
    return this.http.put<Salle>(`${this.apiUrl}/${id}`, salle);
  }

  delete(id: string): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
